cookbooks/travis_build_environment/recipes/hostname.rb

```diff
@@ -52,7 +52,8 @@
   group 'root'
   mode 0o644
   variables(hostname: hostname)
-  only_if { node['travis_build_environment']['update_hosts'] }
+  only_if node['travis_build_environment']['update_hosts']
```

Comment:
This condition check needs to stay in a block if you want it to resolve consistently.